sassoftware/vscode-sas-extension
# fix: reset ready timeout on successful connection

**Summary**
Correct the connection ready timeout to clear when a successful connection is made. The previous behavior was prematurely closing the ssh session because the timeout was firing even after a successful connection ready event was encountered. In the timeout function there is code to gracefully end the connection.

This behavior resulted in subsequent run button presses resetting/re-establishing the sas -nodms interactive session. When this happened any intermediate work table content was lost.

**Testing (Developer)**
- [x] Test via steps to reproduce:
        1. Connect to a 9.4 environment with a SAS 9.4 Remote connection profile.
        2. Place the following sas code in the editor contents:
            `data test; set sashelp.class; run;` 
        3. Click run. Verify that no errors appear in the log.
        4. Replace editor contents with code that uses the work table:
            `proc print data=test; run;`
        5. Click run.        
        6. Verify that proc print step from 4 prints the results from the work table to the log.
        7. Verify that result tab shows with 19 observations.
        8. Click the close button.
        9. Replace the editor contents with code from step 4.
        10. Click run.
        11. Verify that the intermediate work table is no longer found with error:
        `File WORK.TEST.DATA does not exist.`

- [x] Test that a sas program that runs for longer than the ready timeout (default 10) seconds value does not prematurely timeout. Program code:
      `data _null_;
          slept=sleep(25,1);
      run;`
- [x] Add appropriate unit tests
